parsl/monitoring/radios/udp.py

```diff
@@ -0,0 +1,56 @@
+import logging
+import pickle
+import socket
+
+from parsl.monitoring.radios.base import MonitoringRadioSender
+
+
+class UDPRadioSender(MonitoringRadioSender):
+
+    def __init__(self, monitoring_url: str, timeout: int = 10):
+        """
+        Parameters
+        ----------
+
+        monitoring_url : str
+            URL of the form <scheme>://<IP>:<PORT>
+        timeout : int
+            timeout, default=10s
+        """
+        self.monitoring_url = monitoring_url
+        self.sock_timeout = timeout
+        try:
+            self.scheme, self.ip, port = (x.strip('/') for x in monitoring_url.split(':'))
+            self.port = int(port)
```

Comment:
I suppose simpler than using `urllib.parse`.

Reply:
it's probably better to use urllib. I've been hoping to delete the UDP radio entirely though so I've been concentrating on figuring out how to arrange things to put in an ok-ish replacement rather than on making it nicer.